fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java

```diff
@@ -319,10 +320,8 @@ public String toString() {
             builder.append("#").append(ownerGroup.getGroupId().asInt());
         }
 
-        DecimalFormat decimalFormat = new DecimalFormat();
-        decimalFormat.setGroupingSize(3);
-        builder.append(" cost=").append(decimalFormat.format((long) cost));
-        builder.append(" estRows=").append(estOutputRowCount);
+        builder.append(" cost=").append(NUM_FORMAT.format((long) cost));
+        builder.append(" estRows=").append(NUM_FORMAT.format(estOutputRowCount));
```

Comment:
please add synchronized block for this code, because the NUM_FORMAT is a global variable

<img width="726" alt="image" src="https://user-images.githubusercontent.com/8806055/225260199-bb2d4f0d-ac96-489e-84c6-ce82b9e8334e.png">